files/unarchive.py

```diff
@@ -627,7 +642,10 @@ def is_unarchived(self):
         # setting anyway
 
         # What is different
-        unarchived = True
+        if rc == 0:
```

Comment:
I am not convinced that this is correct. A non-zero return code means nothing here. And certainly should not cause unextracting the archive.